### DISCUSSION TITLE
My builds are failing because the file .../Release/file_opts is missing.

### DISCUSSION THREAD

**Discussion by emintz** [Sloeber/arduino-eclipse-plugin#1576] [Q&A]
Hi, Everyone, I hope that someone can help me with this.

After I installed version 2.0.10 of the ESP32 library, all my builds fail with the following error:

  xtensa-esp32-elf-g++: error: /home/eric/Workspaces/temp/Blinky/Release/file_opts: No such file or directory

There don't seem to be any matching discussions or issues.

I have verified this on an empty project in a newly created workspace.

file_opts is configured in the following files:

  ~/Sloeber/arduinoPlugin/packages/esp32/hardware/esp32/2.0.10/platform.sloeber.txt
  ~/Sloeber/arduinoPlugin/packages/esp32/hardware/esp32/2.0.10/platform.txt

Note that I have installed Sloeber in my home directory.

I've opened a bug report, but if someone could please advise, I'd be deeply grateful and perhaps I could save the Dev team some work.

I absolutely love Sloeber and I hope that I can fix this, or maybe contribute a fix if I can.

Thank you.

**Comment by der-Lemmi** [comment]
in platform.txt you can add

recipe.hooks.prebuild.7.pattern=bash -c "[ -f "{build.path}"/file_opts ] || touch "{build.path}"/file_opts"
recipe.hooks.prebuild.7.pattern.windows=cmd /c if not exist "{build.path}\file_opts" type nul > "{build.path}\file_opts"

this is my workaround

**Reply by jantje** [reply]
here his the process to get the .txt changes into you projects https://github.com/Sloeber/arduino-eclipse-plugin/discussions/1533

**Comment by emintz** [comment]
That's what I did. Works a treat!

**Reply by uzi18** [reply]
@jantje please add to 1533 topic info about transition from/ to specific version

**Reply by jantje** [reply]
It is not clear to me what you are asking here.

**Reply by uzi18** [reply]
@jantje sorry for miss understanding, was thinking about configuration transition between versions of Sloeber. But this is about something different.

**Reply by jantje** [reply]
No problem.
But the latest sloeber version has a share in version control that creates a sloeber.cfg for that purpose.
I'm not sure I'm happy with that one right now as the sloeber.cfg contains all configurations and maybe having a file per config is better.

**Comment by emintz** [comment]
+1 for configuration-specific files. It would have helped me diagnose my
problem much more easily.

Just my opinion, for what it's worth. Thank you so much!

-- Eric

**Reply by jantje** [reply]
Please share your opinion here:
https://github.com/Sloeber/arduino-eclipse-plugin/discussions/1583
